### ISSUE TITLE
`is` operator affects null-state of the operand

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#29318]
```C#
        interface IBase { }
        interface IDerived : IBase { }
        void M(IBase x)
        {
            x.ToString();
            if (x is IDerived)
            {
                x.ToString();                
            }
            else
            {
                x.ToString(); // unexpected warning
            }
            x.ToString(); // unexpected warning
        }
```

![image](https://user-images.githubusercontent.com/12466233/44163438-4e997680-a078-11e8-9188-603da1a5a184.png)

FYI @cston 

Found in dogfood

**Comment by jcouv** [comment]
Fix for false warning is merged (will be available in dev16 preview1 at the latest)

Filed follow-up issue for refining analysis (https://github.com/dotnet/roslyn/issues/30297)
